### PR TITLE
[CPM-75] API Tests for Courses

### PR DESCRIPTION
**Pull request by Richardng12**
- Created API Tests for Courses
- Edited PUT course API method to send the updated course, rather than the old course
- Changed connectToMongoose to only connect when starting up docker, and not when running tests (temp workaround)
- Changed course-route to be more consistent on .json() and .send()

Sorry for the one commit :(

### CODE REVIEW

**Review by wryao64 — APPROVED** [review]

File: api/src/loaders/mongoose.js [hunk 1]
```diff
@@ -4,21 +4,24 @@ const connectionString = process.env.DB_CONN_URL;
...
 const connectToMongoose = async () => {
-    await mongoose.connect(connectionString, {
-        useNewUrlParser: true,
-        useUnifiedTopology: true,
-        useCreateIndex: true,
-        useFindAndModify: false,
-    });
-    const db = mongoose.connection;
+    //temp workaround to not connect to local db when testing
+    if (process.env.NODE_ENV !== "test") {
```
Comment (daltonlim) [inline api/src/loaders/mongoose.js:8]: Do you need to update programme tests to have this field?

Comment (Richardng12) [reply api/src/loaders/mongoose.js:8]: only in the programme-route-tests which has not been created yet. Those tests will be created in another PR

Comment (daltonlim) [reply api/src/loaders/mongoose.js:8]: ah ok

**Review by daltonlim — APPROVED** [review]
LGTM

**Pull request merged**